lntest/itest/lnd_onchain_test.go

```diff
@@ -301,353 +274,290 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
-	// we can swap the output address.
-	aliceAnchorTxIn := func() wire.TxIn {
-		sweepCopy := aliceAnchor.SweepTx.Copy()
-		for _, txIn := range sweepCopy.TxIn {
-			if txIn.PreviousOutPoint == aliceAnchor.OutPoint {
-				return *txIn
-			}
-		}
-
-		t.Fatalf("anchor op not found")
-		return wire.TxIn{}
-	}()
-
-	// We'll set the signature on the input to nil, and then set the
-	// sequence to 16 (the anchor CSV period).
-	aliceAnchorTxIn.Witness[0] = nil
-	aliceAnchorTxIn.Sequence = anchorCsv
-
-	minerAddr, err := net.Miner.NewAddress()
-	if err != nil {
-		t.Fatalf("unable to get miner addr: %v", err)
-	}
-	addrScript, err := txscript.PayToAddrScript(minerAddr)
-	if err != nil {
-		t.Fatalf("unable to gen addr script: %v", err)
-	}
-
-	// Now that we have the txIn, we can just make a new transaction that
-	// uses a different script for the output.
-	tx := wire.NewMsgTx(2)
-	tx.AddTxIn(&aliceAnchorTxIn)
-	tx.AddTxOut(&wire.TxOut{
-		PkScript: addrScript,
-		Value:    anchorSize - 1,
-	})
-
-	return btcutil.NewTx(tx)
+	assertNumTxInAndTxOut(sweepTx, 4, 1)
 }
 
 // testAnchorThirdPartySpend tests that if we force close a channel, but then
 // don't sweep the anchor in time and a 3rd party spends it, that we remove any
 // transactions that are a descendent of that sweep.
-func testAnchorThirdPartySpend(net *lntest.NetworkHarness, t *harnessTest) {
+func testAnchorThirdPartySpend(ht *lntemp.HarnessTest) {
 	// First, we'll create two new nodes that both default to anchor
 	// channels.
 	//
 	// NOTE: The itests differ here as anchors is default off vs the normal
 	// lnd binary.
 	args := nodeArgsForCommitType(lnrpc.CommitmentType_ANCHORS)
-	alice := net.NewNode(t.t, "Alice", args)
-	defer shutdownAndAssert(net, t, alice)
+	alice := ht.NewNode("Alice", args)
+	defer ht.Shutdown(alice)
 
-	bob := net.NewNode(t.t, "Bob", args)
-	defer shutdownAndAssert(net, t, bob)
+	bob := ht.NewNode("Bob", args)
+	defer ht.Shutdown(bob)
 
-	ctxb := context.Background()
-	net.ConnectNodes(t.t, alice, bob)
+	ht.EnsureConnected(alice, bob)
 
 	// We'll fund our Alice with coins, as she'll be opening the channel.
-	// We'll fund her with *just* enough coins to open the channel.
+	// We'll fund her with *just* enough coins to open the channel and
+	// sweep the anchor.
 	const (
 		firstChanSize   = 1_000_000
 		anchorFeeBuffer = 500_000
 	)
-	net.SendCoins(t.t, firstChanSize, alice)
-
-	// We'll give Alice another spare UTXO as well so she can use it to
-	// help sweep all coins.
-	net.SendCoins(t.t, anchorFeeBuffer, alice)
+	ht.FundCoins(firstChanSize+anchorFeeBuffer, alice)
 
 	// Open the channel between the two nodes and wait for it to confirm
 	// fully.
-	aliceChanPoint1 := openChannelAndAssert(
-		t, net, alice, bob, lntest.OpenChannelParams{
+	aliceChanPoint1 := ht.OpenChannel(
+		alice, bob, lntemp.OpenChannelParams{
 			Amt: firstChanSize,
 		},
 	)
 
+	// Send another UTXO if this is a neutrino backend. When sweeping
```

Comment:
fyi @guggero as I remember we've both tried to debug this area before. I think our anchor sweep logic can be modified a bit to not always create the remote anchor sweep if we know it's a local force close.